### DISCUSSION TITLE
Koishi 4.5.1

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#596] [General]
## Features

- **bind:** 支持了 `tokenPrefix` 配置项 (#591) (ed811b53d7a28cde303fdcd909308ea7bd5c95ef)
- **core:** 初步支持了 i18n formatter 语法 (94e533ad639dfc2f6cff86ba4b201698f1e12761)
- **switch:** 加入了多语言翻译文件 (f1f7c3f737e77dddd63f5349a6f45d8d027bb9dd)

## Bug Fixes

- **core:** 指令名不应该在运行时被更改 (#574) (2c5d54b9e16145f632279eebdb20a3bf0646810f)
- **discord:** 修复了 `bot.getMessage()` 部分情况下 `content` 为空串的问题 (#585) (08891abc9111f31b671bcd379367f27dea3070d8)
- **discord:** 修复了部分 segment 未被正确转换的问题 (#583) (4442a07feba9fa811053930c80113a6b8e577f85)
- **discord:** 修复了下载的资源缺少后缀名导致无法识别的问题 (#569) (b4dd6e614434679881cc59805f723b62b6059f16)
- **mongo:** 修复了 `database.create()` 中的错误会在之后的调用被抛出的问题 (#589) (30d89b685bb2f7e42e95b8f53bfb194c14c2c55c)
- **mongo:** 修复了控制台显示数据库未安装的问题 (#568) (#592) (1fad834fc03115eb797bd32ffae144536f589275)
- **puppeteer:** 修复了配置项错误的问题 (#593) (#594) (868681dbe2671d26ae9ca08639c333378a49a661)
- **scripts:** 修复了 new 指令无法正确获取输入的插件名的问题 (#588) (93d855ccce6d8e9c3fe14a08f2b3c18c62c039c7)

## Other Changes

- docs: fix code typings (4ccc7eb9a29cc7d84917128791f961382d860d8a)
- docs: fix code typings (a55e7a92ec2099bec2104baf31e2dec164b44510)
- docs: add context state api (e846fdbf14f029b68beb549b374ff27a53b9abb7)
- docs: add service api (15bb8ba9d48aea26dfd4b4d559322c6bc0d4755d)
- docs: fix code typings (47ba4098f97b03c9486060b7bfc0995f668cf2c1)
- docs: fix code typings (4a10e7282ffd16a8c1613f9f2e52cb70e83bb40b)
- chore: fix lint (5ea0e79a074dc0e1a55d5afbe8a838692978ceba)
- docs: enhance database docs (8becc4d6280fa7580afc0590ece0798e0a77e6ce)
- docs: add more database examples (c5c5c7575bd84a8c366d0cf790db3124a321c592)
- chore: bump versions (b96930bfbb542f75ce9acc1b10c0df5829e9e76e)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.5.1'>Koishi 4.5.1</a>.</em>

**Comment by MaikoTan** [comment]
🎉 Hooray! 🎉
